maven-core/src/main/java/org/apache/maven/plugin/version/internal/DefaultPluginVersionResolver.java

```diff
@@ -90,22 +94,35 @@
     public PluginVersionResult resolve( PluginVersionRequest request )
         throws PluginVersionResolutionException
     {
-        PluginVersionResult result = resolveFromProject( request );
+        Map<String, PluginVersionResult> cache = getCache( request.getRepositorySession().getData() );
+        String key = getKey( request );
+
+        PluginVersionResult result = cache.get( key );
```

Comment:
This is get, but where do you populate the cache?

Reply:
Ouch :-( it must have been removed during the refactoring...